src/app/component/pokemons/pokemons.component.ts

```diff
@@ -0,0 +1,50 @@
+import { Component } from '@angular/core';
+import { Pokemon } from '../../model/pokemon';
+import { Router } from '@angular/router';
+import { PokemonService } from 'src/app/services/pokemon.service';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-pokemons',
+  templateUrl: './pokemons.component.html',
+  styleUrls: ['./pokemons.component.scss']
+})
+export class PokemonsComponent {
+  pokemons: Pokemon[] = [];
+  searchText: string = "";
+  filteredPokemons: Pokemon[] = [];
+  pokemonsSubscription: Subscription | undefined;
+
+  constructor(private router: Router, private pokService: PokemonService) {}
+
+  ngOnInit(): void {
+    this.pokemonsSubscription = this.pokService.getPokemons().subscribe((val: Pokemon[]) => {
+      this.pokemons = val;
```

Comment:
if this.pokemons without using remove it

Reply:
I'm using this to keep the state of the original array between filtering to filtering